### PR TITLE
Added multivariate forms of Laplace, studentT and NormalGamma distributions

### PR DESCRIPTION
**Pull request by akash9712**
Added 3 new types of multivariate distributions. `MultivariateLaplace`
distribution can be called using `Laplace` by using lists/Matrices
instead of real numbers. Multivariate StudentT distribution is called
using `MultivariateT` and not using `StudentT` because of difference in
both type and number of parameters. NormalGamma distribution is called
using `NormalGamma`. Added and modified tests accordingly.

### CODE REVIEW

**Comment by akash9712** [comment]
@Upabjojr @jksuom Please have a look.

File: sympy/stats/joint_rv_types.py [hunk 1]
```diff
@@ -58,3 +58,152 @@ def marginal_distribution(self, indices, sym):
...
+
+    def pdf(self, *args):
+        from sympy.functions.special.bessel import besselk
+        mu, sigma = Matrix(self.mu), Matrix(self.sigma)
+        k = len(mu)
+        sigma_inv = sigma**(-1)
+        args = Matrix(args)
+        v = 1 - S(k)/2
+        return S(2)/((2*pi)**(S(k)/2)*sqrt(det(sigma))) \
+        *((args.transpose()*sigma_inv*args)[0]/(2 + (mu.transpose()*sigma_inv*mu)[0])) \
```
Comment (jksuom) [inline sympy/stats/joint_rv_types.py]: The returned expression would be more readable (and more efficient) if `args.transpose()` and `mu.transpose()` were precomputed and assigned to local variables. Even `(mu.transpose()*sigma_inv*mu)[0]` could be precomputed.

File: sympy/stats/joint_rv_types.py [hunk 2]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+    def set(self):
+        k = len(self.mu)
+        return S.Reals**k
+
+    def check(self, mu, sigma):
+        mu, sigma = Matrix([mu]), Matrix(sigma)
+        _value_check(len(mu) == len(sigma.col(0)),
+            "Size of the mean vector and covariance matrix are incorrect.")
+        #check if covariance matrix is positive definite or not.
+        _value_check(all([i > 0 for i in sigma.eigenvals().keys()]),
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: what about the case in which the eigenvalues cannot be computed?

File: sympy/stats/joint_rv_types.py [hunk 3]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+            "The covariance matrix must be positive definite. ")
+
+    def pdf(self, *args):
+        from sympy.functions.special.bessel import besselk
+        mu, sigma = Matrix(self.mu), Matrix(self.sigma)
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: `Matrix` not needed here, `self.mu` and `self.sigma` are already supposed to be matrices.

File: sympy/stats/joint_rv_types.py [hunk 4]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+        mu_T = mu.transpose()
+        k = len(mu)
+        sigma_inv = sigma**(-1)
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: `sigma.inv()` maybe is clearer

File: sympy/stats/joint_rv_types.py [hunk 5]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+        args = Matrix(args)
+        args_T = args.transpose()
+        v = 1 - S(k)/2
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: `S(k)` gets repeated

File: sympy/stats/joint_rv_types.py [hunk 6]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+        return S(2)/((2*pi)**(S(k)/2)*sqrt(det(sigma))) \
+        *((args_T*sigma_inv*args)[0]/(2 + (mu_T*sigma_inv*mu)[0])) \
+        **(S(v)/2)*besselk(v, sqrt((2 + (mu_T*sigma_inv*mu)[0])
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: some repeated expressions could be pre-computed, like ` (mu_T*sigma_inv*mu)[0]`

File: sympy/stats/joint_rv_types.py [hunk 7]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+    def set(self):
+        k = len(self.mu)
+        return S.Reals**k
+
+    def check(self, mu, sigma, v):
+        mu, sigma = Matrix([mu]), Matrix(sigma)
+        _value_check(len(mu) == len(sigma.col(0)),
+            "Size of the location vector and shape matrix are incorrect.")
+        #check if covariance matrix is positive definite or not.
+        _value_check(all([i > 0 for i in sigma.eigenvals().keys()]),
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: eigenval problem?

File: sympy/stats/joint_rv_types.py [hunk 8]
```diff
@@ -58,3 +58,154 @@ def marginal_distribution(self, indices, sym):
...
+            "The shape matrix must be positive definite. ")
+
+    def pdf(self, *args):
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: same reasoning

**Comment by Upabjojr** [comment]
Do `variance( )`, `expectation( )` and `probability( )` already work with multivariate RV? If yes, it would be nice to have some tests.

File: sympy/stats/joint_rv_types.py (hunk 3 above)
Comment (akash9712) [reply sympy/stats/joint_rv_types.py]: `self.sigma` and `self.mu` may also be of  type `list`, so `Matrix` just makes sure they are transformed into matrices.

**Comment by akash9712** [comment]
@Upabjojr I have made the changes you mentioned in review comments. `variance( )`, `expectation( )` and `probability( )` do not yet work for these. I will make a seperate PR for them, if that's okay.

File: sympy/stats/joint_rv_types.py (hunk 3 above)
Comment (Upabjojr) [reply sympy/stats/joint_rv_types.py]: They should be stored as immutable matrices, the transformation should not happen here.

**Comment by Upabjojr** [comment]
> variance( ), expectation( ) and probability( ) do not yet work for these. I will make a seperate PR for them, if that's okay.

OK

**Pull request merged**